wgpu-types/src/lib.rs

```diff
@@ -2109,6 +2037,467 @@ pub enum TextureFormat {
+                    "rgba16snorm" => TextureFormat::Rgba16Snorm,
+                    "rgba16float" => TextureFormat::Rgba16Float,
+                    "rgba32uint" => TextureFormat::Rgba32Uint,
+                    "rgba32sint" => TextureFormat::Rgba32Sint,
+                    "rgba32float" => TextureFormat::Rgba32Float,
+                    "depth32float" => TextureFormat::Depth32Float,
+                    "depth32float-stencil8" => TextureFormat::Depth32FloatStencil8,
+                    "depth24plus" => TextureFormat::Depth24Plus,
+                    "depth24plus-stencil8" => TextureFormat::Depth24PlusStencil8,
+                    "depth24unorm-stencil8" => TextureFormat::Depth24UnormStencil8,
+                    "rgb9e5ufloat" => TextureFormat::Rgb9e5Ufloat,
+                    "bc1-rgba-unorm" => TextureFormat::Bc1RgbaUnorm,
+                    "bc1-rgba-unorm-srgb" => TextureFormat::Bc1RgbaUnormSrgb,
+                    "bc2-rgba-unorm" => TextureFormat::Bc2RgbaUnorm,
+                    "bc2-rgba-unorm-srgb" => TextureFormat::Bc2RgbaUnormSrgb,
+                    "bc3-rgba-unorm" => TextureFormat::Bc3RgbaUnorm,
+                    "bc3-rgba-unorm-srgb" => TextureFormat::Bc3RgbaUnormSrgb,
+                    "bc4-r-unorm" => TextureFormat::Bc4RUnorm,
+                    "bc4-r-snorm" => TextureFormat::Bc4RSnorm,
+                    "bc5-rg-unorm" => TextureFormat::Bc5RgUnorm,
+                    "bc5-rg-snorm" => TextureFormat::Bc5RgSnorm,
+                    "bc6h-rgb-ufloat" => TextureFormat::Bc6hRgbUfloat,
+                    "bc6h-rgb-float" => TextureFormat::Bc6hRgbSfloat,
+                    "bc7-rgba-unorm" => TextureFormat::Bc7RgbaUnorm,
+                    "bc7-rgba-unorm-srgb" => TextureFormat::Bc7RgbaUnormSrgb,
+                    "etc2-rgb8unorm" => TextureFormat::Etc2Rgb8Unorm,
+                    "etc2-rgb8unorm-srgb" => TextureFormat::Etc2Rgb8UnormSrgb,
+                    "etc2-rgb8a1unorm" => TextureFormat::Etc2Rgb8A1Unorm,
+                    "etc2-rgb8a1unorm-srgb" => TextureFormat::Etc2Rgb8A1UnormSrgb,
+                    "etc2-rgba8unorm" => TextureFormat::Etc2Rgba8Unorm,
+                    "etc2-rgba8unorm-srgb" => TextureFormat::Etc2Rgba8UnormSrgb,
+                    "eac-r11unorm" => TextureFormat::EacR11Unorm,
+                    "eac-r11snorm" => TextureFormat::EacR11Snorm,
+                    "eac-rg11unorm" => TextureFormat::EacRg11Unorm,
+                    "eac-rg11snorm" => TextureFormat::EacRg11Snorm,
+                    other => {
+                        if other.starts_with("astc") {
+                            let mut parts = other.split('-');
+                            parts.next(); // discard "astc"
+                            let block = parts
+                                .next()
+                                .ok_or_else(|| E::invalid_value(Unexpected::Str(s), &self))?;
+                            let channel = parts.collect::<Vec<&str>>().join("-");
```

Comment:
```suggestion
                        if let Some(parts) = other.strip_prefix("astc-") {
                            let (block, channel) = parts
                                .split_once('-')
                                .ok_or_else(|| E::invalid_value(Unexpected::Str(s), &self))?;
```

I think using `strip_prefix("astc-")` and then `split_once('-')` would be clearer here. (It could be `split` and `Split::as_str` but the latter is unstable :'( )

Additionally, it would avoid two allocations and would discard invalid formats such as "astcXXX-4x4-unorm"